common/lib/common-utils/src/idleTaskScheduler.ts

```diff
@@ -0,0 +1,35 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
+ * API used to schedule non-essential tasks
+ * Time measurements are in milliseconds as a floating point with a decimal
+ * Takes in and runs a callback during idle time. Fallback to setTimeout if window doesn't
+ * support requestIdleCallback.
+ * @returns A promise pertaining to the callback that was passed in.
+ */
+export async function scheduleIdleTask<T>(callback: () => T, timeout: number): Promise<T> {
+    // Check for the availability in window.
+    return typeof window?.requestIdleCallback === "function"
+        ? new Promise<T>((resolve, reject) => {
+              function doLowPriorityTask(deadline): void {
```

Comment:
I've shared feedback before - please use 
const doLowPriorityTask = () => {
...
};

and share code between different code paths - given that deadline arg is not used, you can use same doLowPriorityTask in both cases.